tests/fixtures/optimizer/canonicalize.sql

```diff
@@ -2,7 +2,7 @@ SELECT w.d + w.e AS c FROM w AS w;
 SELECT CONCAT("w"."d", "w"."e") AS "c" FROM "w" AS "w";
 
 SELECT CAST(w.d AS DATE) > w.e AS a FROM w AS w;
-SELECT CAST("w"."d" AS DATE) > CAST("w"."e" AS DATETIME) AS "a" FROM "w" AS "w";
+SELECT CAST("w"."d" AS DATE) > CAST("w"."e" AS DATE) AS "a" FROM "w" AS "w";
```

Comment:
how does this work? i thought only mysql does this

Reply:
I think my other [comment](https://github.com/tobymao/sqlglot/pull/4431#discussion_r1853460865) also answers this. Let me know if it needs further clarification, though.